### PR TITLE
updates to test data page

### PR DESCRIPTION
**Pull request by rosieyohannan**
* Bring test data page in line with other data persistence pages
* Pros of test results and artifacts for test data storage
* Include latest insights options for cloud and server 3
* Update endpoint for metadata collection
* remove store artifacts steps from examples
* link out to persisting data page for storage usage info

To do - once persisting data and optimisations changes are in, I will add "test results" to the persisting data overview and add relevant links

### CODE REVIEW

File: jekyll/_cci2/collect-test-data.md [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
-You can access the test results interface from the *Tests* tab when viewing any particular [job]({{ site.baseurl}}/2.0/concepts/#jobs),
-as seen below.
+You can access the test results from the **Tests** tab when viewing a job, as shown below.
 
 ![store-test-results-view]( {{ site.baseurl }}/assets/img/docs/test-summary.png)
 
-To see test results as build artifacts, upload them using the [`store_artifacts`]({{ site.baseurl}}/2.0/configuration-reference/#store_artifacts) step.
-
-The usage of the [`store_test_results`]({{ site.baseurl}}/2.0/configuration-reference/#store_test_results) key in your config looks like the following:
+Below is an example of using  the [`store_test_results`]({{ site.baseurl}}/2.0/configuration-reference/#store_test_results) key in your `.circleci/config.yml`:
```
Comment (devinmarieb) [inline jekyll/_cci2/collect-test-data.md]:
```suggestion
Below is an example of using the [`store_test_results`]({{ site.baseurl}}/2.0/configuration-reference/#store_test_results) key in your `.circleci/config.yml`:
```

File: jekyll/_cci2/collect-test-data.md [hunk 2]
```diff
@@ -1,47 +1,63 @@
...
     path: test-results
 ```
 
-Where the `path` key is an absolute or relative path to your `working_directory` containing subdirectories of JUnit XML or Cucumber JSON test metadata files, or the path of a single file containing all test results. Make sure that your `path` value is not a hidden folder (example: `.my_hidden_directory` would be an invalid format).
+The `path` key is an absolute or relative path to your `working_directory` containing subdirectories of JUnit XML or Cucumber JSON test metadata files, or the path of a single file containing all test results.
+
+**Note:** Make sure that your `path` value is not a hidden folder. For example, `.my_hidden_directory` would be an invalid format.
+
+## Test Insights
+See the [Test Insights guide]({{site.baseurl}}/2.0/insights-tests/) for information on using the Insights feature to gather information about your tests, including flaky test detection, viewing alist of tests that fail most often, slowest tests and abn overall performance summary.
```
Comment (devinmarieb) [inline jekyll/_cci2/collect-test-data.md:57]: Should insights be capitalized?

File: jekyll/_cci2/collect-test-data.md [hunk 3]
```diff
@@ -1,47 +1,63 @@
...
 * TOC
 {:toc}
 
+## Overview
+
+Using the **`store_test_results` step** gives you access to:
+
+* The **Tests** pane in the CircleCI web app, as described in the {{site.url}}/{{site.baseurl}}/2.0/collect-test-data/[Collecting Test Metadata guide].
+* Test insights and flaky test detection.
+* Test splitting
```
Comment (devinmarieb) [inline jekyll/_cci2/collect-test-data.md:17]: I think we should be consistent in a bulleted list if we're using full stops or not after each item.

File: jekyll/_cci2/collect-test-data.md [hunk 4]
```diff
@@ -18,6 +18,8 @@ Using the **`store_test_results` step** gives you access to:
 
 Alternatively, storing test results as **artifacts** means you can look at the raw xml. This can be useful when debugging issues with setting up your project's test results handling, for example, working out if you are uploading incorrect files. To see test results as build artifacts, upload them using the [`store_artifacts` step ]({{ site.baseurl}}/2.0/configuration-reference/#store_artifacts).
 
+**Note:** You might choose to upbload your test results using both `store_test_results` and `store_artifacts`
```
Comment (devinmarieb) [inline jekyll/_cci2/collect-test-data.md:21]:
```suggestion
**Note:** You might choose to upload your test results using both `store_test_results` and `store_artifacts`.
```

**Review by devinmarieb — COMMENTED** [review]
Just a few comments, looks great!

File: jekyll/_cci2/collect-test-data.md (hunk 3 above)
Comment (rosieyohannan) [reply jekyll/_cci2/collect-test-data.md:17]: Oh yeah ooops

File: jekyll/_cci2/collect-test-data.md [hunk 5]
```diff
@@ -1,47 +1,70 @@
...
 ---
 
-CircleCI collects test metadata from XML files and uses it to provide insights into your job. This document describes how to configure CircleCI to output test metadata as XML for some common test runners and store reports with the `store_test_results` step.
+When you run tests in CircleCI there two ways to store your test results. You can use either [artifacts]({{site.baseurl}}/2.0/artifacts) or the [`store_test_results` step]({{site.baseurl}}/2.0/configuration-reference/#storetestresults). There are advantages to both methods, so the decision needs to be made for each project. When you save test data using the `store_test_results` step, CircleCI collects data from XML files and uses it to provide insights into your job. This page describes how to configure CircleCI to output test data as XML for some common test runners and store reports with the `store_test_results` step.
+
+Using the **`store_test_results` step** gives you access to:
+
+* The **Tests** pane in the CircleCI web app.
+* Test insights and flaky test detection.
+* Test splitting
```
Comment (rosieyohannan) [inline jekyll/_cci2/collect-test-data.md:17]:
```suggestion
* Test splitting.
```

File: jekyll/_cci2/collect-test-data.md [hunk 6]
```diff
@@ -1,47 +1,70 @@
...
+
+Alternatively, storing test results as **artifacts** means you can look at the raw xml. This can be useful when debugging issues with setting up your project's test results handling, for example, working out if you are uploading incorrect files. To see test results as build artifacts, upload them using the [`store_artifacts` step ]({{ site.baseurl}}/2.0/configuration-reference/#store_artifacts).
+
+**Note:** You might choose to upbload your test results using both `store_test_results` and `store_artifacts`
 
 * TOC
 {:toc}
 
+## Overview
```
Comment (rosieyohannan) [inline jekyll/_cci2/collect-test-data.md:26]:
```suggestion
## Overview
{: #overview }
```

File: jekyll/_cci2/collect-test-data.md [hunk 7]
```diff
@@ -1,47 +1,70 @@
...
+  - store_test_results:
     path: test-results
 ```
 
-Where the `path` key is an absolute or relative path to your `working_directory` containing subdirectories of JUnit XML or Cucumber JSON test metadata files, or the path of a single file containing all test results. Make sure that your `path` value is not a hidden folder (example: `.my_hidden_directory` would be an invalid format).
+The `path` key is an absolute or relative path to your `working_directory` containing subdirectories of JUnit XML or Cucumber JSON test metadata files, or the path of a single file containing all test results.
+
+**Note:** Make sure that your `path` value is not a hidden folder. For example, `.my_hidden_directory` would be an invalid format.
+
+## Viewing storage usage
```
Comment (rosieyohannan) [inline jekyll/_cci2/collect-test-data.md:52]:
```suggestion
## Viewing storage usage
{: #viewing-storage-usage }
```

File: jekyll/_cci2/collect-test-data.md [hunk 8]
```diff
@@ -1,47 +1,70 @@
...
+
+For information on viewing your stoarage usage, and calculating your monthly storage overage costs, if applicable, see the [Persisting Data]({{site.baseurl}}/2.0/persist-data/#managing-network-and-storage-use) guide.
 
+## Test Insights
```
Comment (rosieyohannan) [inline jekyll/_cci2/collect-test-data.md:56]:
```suggestion
## Test Insights
{: #test-insights }
```

File: jekyll/_cci2/collect-test-data.md (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_cci2/collect-test-data.md:57]: I think this is like the odd one out and is capitalised but I will check :-D

Comment (rosieyohannan) [reply jekyll/_cci2/collect-test-data.md:60]: Seems to be a yes

**Review by devinmarieb — APPROVED** [review]

**Pull request merged**